qiskit/providers/aer/noise/errors/quantum_error.py

```diff
@@ -185,79 +315,75 @@ def copy(self):
-                "Invalid atol ({}) must be non-negative.".format(value))
-        if value > cls._MAX_TOL:
-            raise NoiseError(
-                "Invalid atol ({}) must be less than {}.".format(
-                    value, cls._MAX_TOL))
-        cls._ATOL_DEFAULT = value
+        warnings.warn(
+            'QuantumError.set_atol(value) has been deprecated as of qiskit-aer 0.10.0'
+            ' and will be removed no earlier than 3 months from that release date.'
+            ' Use QuantumError.atol = value instead.',
+            DeprecationWarning, stacklevel=2)
+        QuantumError.atol = value
 
     @classmethod
     def set_rtol(cls, value):
         """Set the class default relative tolerance parameter for float comparisons."""
-        if value < 0:
-            raise NoiseError(
-                "Invalid rtol ({}) must be non-negative.".format(value))
-        if value > cls._MAX_TOL:
-            raise NoiseError(
-                "Invalid rtol ({}) must be less than {}.".format(
-                    value, cls._MAX_TOL))
-        cls._RTOL_DEFAULT = value
+        warnings.warn(
+            'QuantumError.set_rtol(value) has been deprecated as of qiskit-aer 0.10.0'
+            ' and will be removed no earlier than 3 months from that release date.'
+            ' Use QuantumError.rtol = value instead.',
+            DeprecationWarning, stacklevel=2)
+        QuantumError.rtol = value
 
     @property
     def size(self):
         """Return the number of error circuit."""
-        return len(self._noise_circuits)
+        return len(self.circuits)
 
     @property
     def number_of_qubits(self):
         """Return the number of qubits for the error."""
-        return self._number_of_qubits
+        warnings.warn(
+            '"number_of_qubits" property has been renamed to num_qubits and deprecated as of'
+            ' qiskit-aer 0.10.0, and will be removed no earlier than 3 months'
+            ' from that release date. Use "num_qubits" instead.',
+            DeprecationWarning, stacklevel=2)
+        return self.num_qubits
 
     @property
     def circuits(self):
         """Return the list of error circuits."""
-        return self._noise_circuits
+        return self._circs
 
     @property
     def probabilities(self):
         """Return the list of error probabilities."""
-        return self._noise_probabilities
+        return self._probs
 
     def ideal(self):
         """Return True if current error object is an identity"""
-        instructions, probability = self.error_term(0)
-        if (probability == 1 and len(instructions) == 1 and
-                instructions[0]["name"] == "id"):
-            logger.debug("Error object is ideal")
-            return True
-        return False
+        for circ in self.circuits:
+            # check if circ is composed of identity gates up to global phase
+            for gate, _, _ in circ:
+                if not isinstance(gate, IGate) and \
```

Comment:
Umm, if we could change the spec, I'd like to simplify the implementation to
```
return is_identity_matrix(self.to_quantumchannel().data, ignore_phase=True, atol=self.atol, rtol=self.rtol))
```
rather than trying to catch all possible cases here. What do you think?
I know my suggested implementation would be "slightly" slower than current `ideal()`, which actually try to implement `has_all_identity_circuits()`, but I believe it is the most precise implementation for `ideal()`.

Reply:
This is going to be inefficient (or impossible to compute) for Pauli errors or clifford error circuits on large numbers of qubits since you need to construct large unitaries have to construct the unitary. This is why I suggest first checking the scalable efficient cases, before the general unitary case in the above code block. I would rather it be a generally useful function, since you can always just do the identity check of `Kraus(qerror)` yourself if you want to check that way.

Reply:
OK, I keep the original spec following your suggestion and update the docstring accordingly. 902f541